### PR TITLE
20210210 article - ZAP & ASVS

### PR DESCRIPTION
**Pull request by BlazingWind**
An article about ZAP ASVS integration

Signed-off-by: BlazingWind <blazingwindsec@gmail.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
To address the DCO requirement you'll need to sign-off the commit(s):

- https://github.com/zaproxy/zaproxy/blob/develop/CONTRIBUTING.md#developer-certificate-of-origin
- https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff

**Review by kingthorin — COMMENTED** [review]
Could you put markdown code fencing on the code blocks and remove the formatting markers?
https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks

File: site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+not to get too much into technical details, to make it easy to
+understand and accessible to everyone.The charset control that we looked
+at earlier has its explanation
+[here](https://github.com/BlazingWind/OWASP-ASVS-4.0-testing-guide/blob/main/14-Configuration/14-4-1-Charset.md).
+
+Does the project sound exciting to you or would you like to contribute?
+I am looking for people that would be willing to write ZAP scripts to
+verify controls, review the controls that are already in the testing
+guide or add new controls to it. Feel free to drop me a message on
+Twitter [\@BlazingWindSec](https://twitter.com/BlazingWindSec)
```
Comment (kingthorin) [inline site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md:325]:
```suggestion
Twitter [&commat;BlazingWindSec](https://twitter.com/BlazingWindSec)
```

File: site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
+---
+title: "Automate checking ASVS controls using ZAP scripts"
+description: "Automate checking ASVS controls using ZAP scripts"
+summary: "Write scripts in ZAP which will check a target application's compliance against ASVS controls."
+type: post
+tags:
+- blog
+date: "2021-02-10"
+addSocialPreview: true
```
Comment (psiinon) [inline site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md:9]: false unless you have a good image to include :)

File: site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+penetration-testable from a black box perspective and this article
+focuses only on level 1 controls. Considering how broad ASVS is, it is
+difficult to ensure that every nook and cranny of an application is
+compliant with ASVS controls.
+
+And here raises a question: how to ensure more coverage of an
+application and make auditing applications with OWASP ASVS easier,
+faster, and more effective?
+
+# Enter OWASP ZAP and Its Scripting Engine
```
Comment (psiinon) [inline site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md:37]: These are better as ## so they appear in the toc

File: site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md (hunk 2 above)
Comment (kingthorin) [reply site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md:9]: I've just asked in slack if ASVS has a logo (I searched quickly and didn't see one).

File: site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+give as good coverage as a ZAP script with a crawler. Additionally, the
+script can be reused for doing periodical controls on an application
+eliminating the need for a person to do it manually. Here we would use a
+passive scan script, because there is no need to modify or send any
+request, but only passively scan the passing traffic. There already
+exists [a rule](https://www.zaproxy.org/docs/alerts/10019/) in
+ZAP that checks if the Content-Type header is missing, but it does not
+check the charset, hence I have written this script.
+
+**import** re
```
Comment (psiinon) [inline site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md:98]: Surround scripts with ``` they look better that way, eg as per https://github.com/zaproxy/zaproxy-website/blob/master/site/content/blog/2020-06-04-zap-2-9-0-highlights/index.md

Comment (BlazingWind) [reply site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md]: Added ``` around scripts in the new commit

File: site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md (hunk 3 above)
Comment (BlazingWind) [reply site/content/blog/2021-02-10-automate-checking-asvs-controls-using-zap-scripts/index.md]: Changed # to ## for all titles in the new commit

**Review by psiinon — APPROVED** [review]
Looks great - thank you!

**Review by kingthorin — APPROVED** [review]
LGTM

**Review by thc202 — APPROVED** [review]

**Pull request merged**